controllers/cluster-registration/cluster_registration_controller.go

```diff
@@ -150,7 +150,7 @@ func (r *RegisteredClusterReconciler) updateRegisteredClusterStatus(regCluster *
 func (r *RegisteredClusterReconciler) getManagedCluster(regCluster *singaporev1alpha1.RegisteredCluster) (clusterapiv1.ManagedCluster, error) {
 	managedClusterList := &clusterapiv1.ManagedClusterList{}
 	managedCluster := clusterapiv1.ManagedCluster{}
-	if err := r.HubClusters[0].Client.List(context.Background(), managedClusterList, client.MatchingLabels{RegisteredClusterNamelabel: regCluster.Name, RegisteredClusterNamespacelabel: regCluster.Namespace}); err != nil {
+	if err := r.HubClusters[0].APIReader.List(context.Background(), managedClusterList, client.MatchingLabels{RegisteredClusterNamelabel: regCluster.Name, RegisteredClusterNamespacelabel: regCluster.Namespace}); err != nil {
```

Comment:
why apireader ?

Reply:
because it is in the HUBInstance and we don't have Client anymore.... maybe a problem of concurrent development and not sync with eachother